pkgs/build-support/dotnet/nuget-to-nix/nuget-to-nix.sh

```diff
@@ -91,10 +92,22 @@ for package in *; do
       exit 1
     fi
 
+    source_independent_hash=""
+    nupkg_file=$(echo "$version/$package.$version.nupkg" | tr '[:upper:]' '[:lower:]')
+    stripped_file=$(mktemp tmp-XXXXXXXX.zip)
+    if zip --quiet "$nupkg_file" --output-file "$stripped_file" --temp-path "$TMPDIR" --delete .signature.p7s ; then
+        source_independent_hash=$(nix-hash --type sha256 --sri "$stripped_file")
+        rm "$stripped_file"
+    else
+        rm "$stripped_file"
+        (( $? == 12 ))
+        source_independent_hash=$(nix-hash --type sha256 --sri "$nupkg_file")
+    fi
+
     if [[ "$source" != https://api.nuget.org/v3/index.json ]]; then
-      echo "  (fetchNuGet { pname = \"$id\"; version = \"$version\"; hash = \"$hash\"; url = \"$url\"; })"
+      echo "  (fetchNuGet { pname = \"$id\"; version = \"$version\"; hash = \"$hash\"; source_independent_hash = \"$source_independent_hash\"; url = \"$url\"; })"
```

Comment:
Note: use `lowerCamelCase` for variable names (see [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/299c0dad477f0102bcbe20448618a3a65b9b08d1/CONTRIBUTING.md#syntax:~:text=Use%20lowerCamelCase%20for%20variable%20names)).

Reply:
Sigh, I got this right when I described what work I was going to do, just not when I did it :(